src/wtmp.c

```diff
@@ -40,7 +40,7 @@ int log_wtmp(const char *line, const char *name, const char *host,
     !(defined(LINUX) || defined(__hpux) || defined (_AIX))
   /* This "auxilliary" utmp doesn't exist under linux. */
 
-#if defined(__sparcv9) && !defined(__NetBSD__) && !defined(__FreeBSD__)
+#if defined(__sun)
```

Comment:
Fair enough.  To be a little more conservative, in terms of not potentially changing behavior for the existing macros, would the following also work for your needs?

    #if (defined(__sparcv9) || defined(_sun)) && !defined(__NetBSD__) && !defined(__FreeBSD__)